### PR TITLE
converting contacts and accounts section to page object model

### PR DESCRIPTION
**Pull request by skristem**
# Critical Changes

# Changes

# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: robot/Cumulus/tests/browser/contacts_accounts/deletebehaviors_household.robot [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
+...             robot/Cumulus/resources/ContactPageObject.py
+...             robot/Cumulus/resources/NPSP.py
+Suite Setup     Run keywords
+...             Open Test Browser
+...             Setup Test Data
 Suite Teardown  Delete Records and Close Browser
 
+***Keywords***
+Setup Test Data
+    &{contact} =              API Create Contact        Email=automation@robot.com
```
Comment (boakley) [inline robot/Cumulus/tests/browser/contacts_accounts/deletebehaviors_household.robot]: It's better to use `@example.com` - that domain is specifically set up for documentation and code examples. robot.com is an actual domain that belongs to somebody.

File: robot/Cumulus/tests/browser/contacts_accounts/deletecontact_household.robot [hunk 2]
```diff
@@ -1,27 +1,37 @@
...
+...             robot/Cumulus/resources/AccountPageObject.py
+...             robot/Cumulus/resources/NPSP.py
+Suite Setup     Run keywords
+...             Open Test Browser
+...             Setup Test Data
 Suite Teardown  Delete Records and Close Browser
 
+***Keywords***
+Setup Test Data
+    &{contact1} =                        API Create Contact    Email=automation@robot.com
```
Comment (boakley) [inline robot/Cumulus/tests/browser/contacts_accounts/deletecontact_household.robot]: I recommend changing this to example.com

File: robot/Cumulus/tests/browser/donations_payments/find_matched_gifts.robot [hunk 3]
```diff
@@ -17,8 +17,7 @@ Find Matching Gifts
     Click Link    link=Show more actions
     Click Link    link=Edit
     Populate Lookup Field    Matching Gift Account    &{Org}[Name]
-    Click Dropdown    Matching Gift Status
-    Click Link    link=Potential
+    Select Value From Dropdown   Matching Gift Status              Potential
     Click Modal Button        Save
     &{contact2} =  API Create Contact    Email=skristem@robot.com
```
Comment (boakley) [inline robot/Cumulus/tests/browser/donations_payments/find_matched_gifts.robot]: I recommend changing this to example.com

File: robot/Cumulus/resources/NPSP.py [hunk 4]
```diff
@@ -278,12 +258,15 @@ def get_id(self):
         self.selenium.get_webelement(locator).click()   
         
         
-    def confirm_value(self, field,value,status):
+    def confirm_field_value(self, field,value,status):
+        """If status is Y then the specified value should be present in the field
+                        N then the specified value should not be present in the field
+        """
```
[the remark this answers is not in the document]
Comment (skristem) [reply robot/Cumulus/resources/NPSP.py]: I really like the idea of contains and does not contain, thanks so much will update this

File: robot/Cumulus/resources/AccountPageObject.py [hunk 5]
```diff
@@ -1,19 +1,17 @@
 from cumulusci.robotframework.pageobjects import ListingPage
 from cumulusci.robotframework.pageobjects import DetailPage
 from cumulusci.robotframework.pageobjects import pageobject
+from Basenpspobjects import BaseNPSPPage
 from NPSP import npsp_lex_locators
 
 @pageobject("Listing", "Account")
-class AccountListingPage(ListingPage):
+class AccountListingPage(ListingPage, BaseNPSPPage):
```
Comment (boakley) [inline robot/Cumulus/resources/AccountPageObject.py]: I really apologize for this, but I gave a bad example. It would be better to change the order of these two. It doesn't actually affect your code in this specific case, but I know people may look at your code as an example and want to copy it. It would be better if we set a good example here.

I recommend  you change this line to look like this:

`class AccountingListingPage(BaseNPSPPage, ListingPage)`

Here's a link to a question on stackoverflow that answers the question of "why". Again, I apologize!

File: robot/Cumulus/resources/Basenpspobjects.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+from robot.libraries.BuiltIn import BuiltIn
```
Comment (boakley) [inline robot/Cumulus/resources/Basenpspobjects.py]: I would like to recommend you rename this file to BaseObjects.py.  The NPSP is redundant, and only having the first letter capitalized is a bit strange.

This isn't a showstopper and it doesn't affect functionality, but this name kind of sticks out from the other files in this folder.

File: robot/Cumulus/resources/AccountPageObject.py (hunk 5 above)
Comment (skristem) [reply robot/Cumulus/resources/AccountPageObject.py]: @boakley I made updates. I dont see the link here can u share it with me again plz

Comment (sgovindankutty) [reply robot/Cumulus/resources/AccountPageObject.py]: @skristem Are you planning to perform the changes as Bryan mentioned above in this PR?

File: robot/Cumulus/resources/AccountPageObject.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    
+@pageobject("Details", "Account")
+class AccountDetailPage(DetailPage):
+    object_name = "Account"
+
+    def _is_current_page(self):
+        """ Verify we are on the Account detail page
+            by verifying that the url contains '/view'
+        """
+        self.npsp.wait_until_url_contains("/view")
```
[the remark this answers is not in the document]
Comment (sgovindankutty) [reply robot/Cumulus/resources/AccountPageObject.py]: +1 to the above comment.

File: robot/Cumulus/resources/NPSP.py [hunk 8]
```diff
@@ -1179,8 +1179,7 @@ def change_view_to(self,view_name):
         self.selenium.wait_until_page_contains(view_name)
 
     def search_field_by_value(self, fieldname, value):
-         """ Populate field with Place Holder as a locator
-             and actual value of the place holder.
+         """ search the field with the placeholder given by 'fieldname' for the given 'value'
```
Comment (sgovindankutty) [inline robot/Cumulus/resources/NPSP.py]: Probably rephrase it to 'Searches the field with the given "fieldname" & "value"' ?

File: robot/Cumulus/resources/NPSP.py [hunk 9]
```diff
@@ -1200,7 +1199,7 @@ def verify_record_is_created_in_database(self,object_name,id):
         """Verifies that a record with specified id is saved 
            in specified object table in database and returns the record"""
         record=self.salesforce.salesforce_get(object_name,id)
-        self.builtin.should_not_be_empty(record)
+        self.builtin.should_not_be_empty(record, msg="the database object {} with id {} is not in the database".format(object_name,id))
```
Comment (sgovindankutty) [inline robot/Cumulus/resources/NPSP.py]: A very trivial change. Consider capitalizing the starting letter of a sentence?

File: robot/Cumulus/resources/AccountPageObject.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (skristem) [reply robot/Cumulus/resources/AccountPageObject.py]: yes I have made this change already

File: robot/Cumulus/resources/AccountPageObject.py (hunk 5 above)
Comment (skristem) [reply robot/Cumulus/resources/AccountPageObject.py]: yes made the change already

File: robot/Cumulus/resources/NPSP.py (hunk 8 above)
Comment (skristem) [reply robot/Cumulus/resources/NPSP.py]: done

File: robot/Cumulus/resources/NPSP.py (hunk 9 above)
Comment (skristem) [reply robot/Cumulus/resources/NPSP.py]: done
